### PR TITLE
[Dev] RN v0.45 Upgrade

### PR DESCRIPTION
**Pull request by sarahscott**
- [x] Update transformer
- [x] Update [various dependencies](https://github.com/artsy/emission/pull/570/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2)
- [x] Update snapshots to accommodate new RN version
- [x] Change all instances of `React.PropTypes` to `PropTypes` from the new standalone package
- [x] Fix Relay compilation error
- [x] Fix `RCTAnimation` subspec error (https://github.com/facebook/react-native/issues/13198)
- [x] Rebase with master
- [x] Fix `SearchResults` test failure in CI
- [x] Stress-test in simulator and on-device <hr data-danger="yep"/>

### Tested on Device?

- [x] @sarahscott

<details>
  <summary>How to get set up with this PR?</summary>
  <p>&nbsp;</p>
   <p><b>To run on your computer:</b></p>
<pre><code>git fetch origin pull/570/head:sarahscott-570-checkout
git checkout sarahscott-570-checkout
yarn install
cd example; pod install; cd ..
open -a Simulator
yarn start</code></pre>
   </p>
   <p>Then run <code>xcrun simctl launch booted net.artsy.Emission</code> once a the simulator has finished booting</p>
   <p><b>To run inside Eigen (prod or beta) or Emission (beta):</b> Shake the phone to get the Admin menu.</p>
   <p>If you see <i>"Use Staging React Env" </i> - click that and restart, then follow the next step.</p>
   <p>Click on <i>"Choose an RN build" </i> - then pick the one that says: "X,Y,Z"</p>
   <p>Note: this is a TODO for PRs, currently  you can only do it on master commits.</p>
</details>

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please assign someone to merge this PR, and optionally include people who should review.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: transformer/upstream.js [hunk 1]
```diff
@@ -22,29 +23,39 @@ const makeHMRConfig = require('babel-preset-react-native/configs/hmr');
 const path = require('path');
 const resolvePlugins = require('babel-preset-react-native/lib/resolvePlugins');
 
-const {compactMapping} = require('react-native/packager/src/Bundler/source-map');
+const {compactMapping} = require('./src/Bundler/source-map');
+
+import type {Plugins as BabelPlugins} from 'babel-core';
```
Comment (sarahscott) [inline transformer/upstream.js]: Here are the offensive `import`s

Comment (alloy) [inline transformer/upstream.js]: Oooh, they are in the updated file. I was going off of the code base in master as-is.

Ok this should be easy. Something like `const BabelPlugins = require("babel-core").Plugins`

**Comment by alloy** [comment]
@sarahscott I’ve cut a release of Relay with the RN 0.45 compatibility release: https://github.com/alloy/relay/releases/download/v0.9.3%2Breact-native-0.45-fix/react-relay-0.9.3.react-native-0.45-fix.tgz

File: src/lib/components/home/artwork_rails/__tests__/artwork_rail-tests.tsx [hunk 2]
```diff
@@ -20,19 +19,23 @@ it("has a min height while data is loading", () => {
...
+  const rail = new ArtworkRail(props) as any
 
-  const instance = wrapper.instance() as ArtworkRail
-  const railStyle = instance.railStyle()
+  /**
+   * This is a less-than-ideal workaround because setState wasn't actually updating the state. Ideally we should bring
+   * back Enzyme or something similar once we figure out compatibility issues with RN 0.45+
+   */
+  rail.state = { didPerformFetch: true }
+
```
Comment (alloy) [inline src/lib/components/home/artwork_rails/__tests__/artwork_rail-tests.tsx]: Oh, this is probably due to `setState` being async, you need to use something like `setImmediate` to perform assertions on the next runloop tick.

File: Example/Podfile [hunk 3]
```diff
@@ -15,7 +15,7 @@ target 'Emission' do
   pod 'Emission', :path => '../'
 
   # As this runs dev, we need the developer web socket
-  pod 'React', :path => react_path, :subspecs => %w(RCTWebSocket)
+  pod 'React', :path => react_path, :subspecs => %w(DevSupport BatchedBridge RCTAnimation)
```
Comment (sarahscott) [inline Example/Podfile]: The `RCTAnimation` subspec causes problems, but I can build Emission [if this postinstall script is included in package.json](https://github.com/facebook/react-native/issues/13198#issuecomment-302917321).

Comment (alloy) [inline Example/Podfile]: Where's the addition to our package.json?

Comment (sarahscott) [inline Example/Podfile]: I'll add it now - wasn't sure if we actually wanted to use it or not.

Comment (alloy) [inline Example/Podfile]: There doesn't appear to be another simpel way around that, no?

Comment (sarahscott) [inline Example/Podfile]: I haven't found one. The `Gene` container relies on `react-native-parallax-scroll-view`, which uses `Animated`. However, the `ArtistCard` also uses `Animated` and works fine without `RCTAnimation`, so perhaps there is a way around it? With the `post-install` script and `RCTAnimation`, though, the `Gene` error is solved, so maybe it's a good idea to include it if we can.

Comment (alloy) [inline Example/Podfile]: `ArtistCard` uses the pure JS animation library, so it’s not using the pod like the `Gene` one is.

**Comment by alloy** [comment]
I can reproduce the CI test failures by running the tests ‘in band’, i.e. `yarn test -- -i`.

**Comment by alloy** [comment]
It happens only in `src/lib/components/consignments/components/__tests__/artist-search-results-tests.tsx`

**Comment by alloy** [comment]
Commenting out [this component](https://github.com/artsy/emission/blob/7f338c82ebe91fd3eb897fed78fd81b965a3f84a/src/lib/components/consignments/components/artist-search-results.tsx#L83-L94) makes it go away. The component is essentially a `TextView`, maybe it needs stubbing? /cc @orta

**Comment by orta** [comment]
That sounds 👍 to me, it's not testing any work from the input itself

**Comment by alloy** [comment]
Can you add a CHANGELOG entry? I think it should then be green, in which case please go ahead and merge 👍

**Comment by sarahscott** [comment]
merging!

**Comment by alloy** [comment]
🍾

File: package.json [hunk 4]
```diff
@@ -4,15 +4,15 @@
...
   "scripts": {
-    "bundle": "react-native bundle --config rn-cli.config.js --platform=ios --dev=false --entry-file=index.ios.js --bundle-output Pod/Assets/Emission.js --sourcemap-output Pod/Assets/Emission.js.map --assets-dest Pod/Assets && rm -r Pod/Assets/assets/node_modules",
+    "bundle": "react-native bundle --config \"$(pwd)\"/rn-cli.config.js --platform=ios --dev=false --entry-file=index.ios.js --bundle-output Pod/Assets/Emission.js --sourcemap-output Pod/Assets/Emission.js.map --assets-dest Pod/Assets && rm -rf Pod/Assets/assets/node_modules",
     "clean-example": "cd Example && xcodebuild -workspace Emission.xcworkspace -scheme Emission -destination 'platform=iOS Simulator,name=iPhone 6' clean",
     "type-check": "tsc --noEmit --pretty",
     "lint": "tslint 'src/**/*.{ts,tsx}'",
     "lint-fix": "npm run lint -- --fix",
     "ci": "npm run type-check && npm run lint && npm run test -- --runInBand",
     "danger": "danger",
-    "start": "npm run clean-example && concurrently --kill-others 'npm run start-packager' 'npm run storybook'",
```
Comment (orta) [inline package.json]: Was there a reason that storybooks were removed from the app? I don't see a mention in the PR body

Comment (orta) [inline package.json]: Shipped https://github.com/artsy/emission/pull/585 to fix it 👍

Comment (alloy) [inline package.json]: Thanks!

**Pull request merged**